loopchain/blockchain/blockchain.py

```diff
@@ -1442,3 +1442,8 @@ def makeup_block(self,
             self.__add_tx_to_block(block_builder)
 
         return block_builder
+
+    def try_update_last_unconfirmed_block(self, unconfirmed_block: 'Block'):
+        """Note that the method expects an input as Block 1.0+"""
+        if not self.__last_block or unconfirmed_block.header.height == self.__last_block.header.height + 2:
```

Comment:
(self.__last_block.header.height + 1) also will need to update.

Reply:
In normal case, no need to request `last_block.header.height + 1` block, because it is candidate block, which makes evidence of `last_block`.

But suppose that there is a node whose last_block height is `last_block.header.height - 1`.
He or she can request `last_block.header.height + 1`.

We will discuss again soon when we deal with sync problems.